src/app/admin/components/users/users.component.ts

```diff
@@ -1,53 +1,59 @@
 import { UserSearchFormValues } from '@admin-types/users/user-search-form-values.type';
-import { AsyncPipe, JsonPipe } from '@angular/common';
-import { Component, inject } from '@angular/core';
+import { AsyncPipe } from '@angular/common';
+import { Component, computed, effect, inject, signal } from '@angular/core';
 import { Router } from '@angular/router';
 import { GovukHeadingComponent } from '@common/govuk-heading/govuk-heading.component';
 import { LoadingComponent } from '@common/loading/loading.component';
+import { FormStateService } from '@services/form-state.service';
 import { UserAdminService } from '@services/user-admin/user-admin.service';
 import { UserService } from '@services/user/user.service';
-import { BehaviorSubject, combineLatest } from 'rxjs';
 import { Subject } from 'rxjs/internal/Subject';
 import { of } from 'rxjs/internal/observable/of';
 import { switchMap } from 'rxjs/internal/operators/switchMap';
 import { tap } from 'rxjs/internal/operators/tap';
 import { UserSearchFormComponent } from './user-search-form/user-search-form.component';
 import { UserSearchResultsComponent } from './user-search-results/user-search-results.component';
 
+const defaultFormValues: UserSearchFormValues = {
+  fullName: null,
+  email: null,
+  userStatus: 'active',
+};
+
 @Component({
   selector: 'app-users',
   standalone: true,
-  imports: [
-    GovukHeadingComponent,
-    UserSearchFormComponent,
-    UserSearchResultsComponent,
-    AsyncPipe,
-    JsonPipe,
-    LoadingComponent,
-  ],
+  imports: [GovukHeadingComponent, UserSearchFormComponent, UserSearchResultsComponent, AsyncPipe, LoadingComponent],
   templateUrl: './users.component.html',
   styleUrl: './users.component.scss',
 })
 export class UsersComponent {
+  private readonly usersSearchKey = 'admin-user-search';
   userService = inject(UserService);
   userAdminService = inject(UserAdminService);
+  formStateService = inject(FormStateService);
+
+  previousformValues = signal(this.formStateService.getFormValues<UserSearchFormValues>(this.usersSearchKey));
+  formValues = computed(() => this.previousformValues() ?? defaultFormValues);
+
   router = inject(Router);
 
   search$ = new Subject<UserSearchFormValues | null>();
   loading$ = new Subject<boolean>();
-  isSubmitted$ = new BehaviorSubject<boolean>(false);
 
-  results$ = combineLatest([this.search$, this.isSubmitted$]).pipe(
+  results$ = this.search$.pipe(
     tap(() => this.startLoading()),
-    switchMap(([values, isSubmitted]) => {
-      if (!values || !isSubmitted) {
-        return of(null);
-      }
+    switchMap((values) => {
+      if (!values) return of(null);
       return this.userAdminService.searchUsers(values);
     }),
     tap(() => this.stopLoading())
   );
 
+  eff = effect(() => {
```

Comment:
Do you need to assign this to a variable?

Reply:
Good question, if I don't store the reference to the effect, it does this:
<img width="704" alt="image" src="https://github.com/user-attachments/assets/738b7274-b827-4b85-862e-20369f34cdb1">
The alternative is to put it in the constructor which now I think about it may be better practice as if all side effects are created there we know where to look.
<img width="338" alt="image" src="https://github.com/user-attachments/assets/4355c95f-e3c9-4450-891f-3d070d3307d6">